springUsers/src/main/java/com/example/springUsers/Controllers/UserController.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.springUsers.Dto.UserDto;
+import com.example.springUsers.Exceptions.CustomAppException;
+import com.example.springUsers.Model.Request.UserRequest;
+import com.example.springUsers.Model.Response.UserResponse;
+import com.example.springUsers.Services.UserServices;
+
+//Main controller that has the routes default route is users
+@RestController
+@RequestMapping("users")
+public class UserController {
+
+	private UserServices userService; 
+	
+	public UserController(UserServices userService) {
+		this.userService = userService;
+	}
+	//post
+	@PostMapping
+	public UserResponse createUser(@RequestBody UserRequest userRequest) {
+		 
+		UserDto userDto = new UserDto(); 
+		BeanUtils.copyProperties(userRequest, userDto);
+		UserDto createdUser = userService.createUser(userDto); 
+		UserResponse returnUser = new UserResponse(); 
+		BeanUtils.copyProperties(createdUser, returnUser);
+		
+		return returnUser; 
+		 
+	}
+	//get
+	@GetMapping
+	public List<UserResponse> getAllUsers(
+			@RequestParam(value="page", defaultValue = "1") int page, 
+			@RequestParam(value="limit", defaultValue="5") int limit) throws CustomAppException{
+		try {
+		List<UserDto> userDtoList = userService.getAllUsers(page, limit);
+		List<UserResponse> userResponseList  = new ArrayList<UserResponse>(); 
+		
+		for ( int i = 0; i<userDtoList.size(); i++ ) {
+			UserResponse userResponse = new UserResponse(); 
+			BeanUtils.copyProperties(userDtoList.get(i), userResponse);
+			userResponseList.add(userResponse); 
+		}
+	
+		return userResponseList; }
+		 catch(NoSuchElementException exc) {
+	            throw new CustomAppException("sorry cant get current list of users");
+	        }
+	}
+	
+	//get
+	@GetMapping(path="/{userId}")
+	public UserResponse getUserByUserId(@PathVariable String userId) throws CustomAppException{
+		try {
+		UserDto singleUserDto = userService.getUserByUserId(userId); 
+		UserResponse returnValue = new UserResponse(); 
+		BeanUtils.copyProperties(singleUserDto, returnValue);
+		return returnValue;  }
+        catch(NoSuchElementException exc) {
+            throw new CustomAppException("That username  doesnt exist");
+        }
+	}
+	
+	//update
+	@PutMapping(path="/{userId}")
+	public UserResponse updateUser(@RequestBody UserRequest userRequest, @PathVariable String userId) {
+		UserDto userDto = new UserDto(); 
+		BeanUtils.copyProperties(userRequest, userDto);
+		userDto.setUserId(userId);
+		UserDto updatedUserDto = userService.updateUser(userDto); 
+		
+		UserResponse returnValue = new UserResponse(); 
+		BeanUtils.copyProperties(updatedUserDto, returnValue);
+		
+		return returnValue; 
+	}
+	//delete
+	
+	@DeleteMapping(path="/{userId}")
+	public void deleteUser(@PathVariable String userId) {
```

Comment:
Have some sort of response for each endpoint. Even if there's no user to respond with, you should return a success message